examples/hotelling_law/app.py

```diff
@@ -0,0 +1,142 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import solara
+from mesa.experimental import JupyterViz
+
+from .model import HotellingModel
+
+
+# This function defines how agents are visually represented in the simulation.
+def agent_portrayal(agent):
+    size = 50  # Default size
+    color = "grey"  # Default color for agents
+
+    # Check if the agent has a 'price' attribute.
+    # This is to ensure compatibility
+    # with different types of agents.
+    if hasattr(agent, "price"):
+        # Adjust color based on the price attribute of the StoreAgent
+        if agent.price > 12:
+            color = "#FF0000"  # Higher prices in red
+        elif agent.price > 8:
+            color = "#FFA500"  # Moderate prices in orange
+        else:
+            color = "#00FF00"  # Lower prices in green
+    # Construct and return the portrayal dictionary
+    portrayal = {
+        "size": size,
+        "color": color,
+    }
+    return portrayal  # Return the portrayal dictionary to be used by
+    # the visualization engine.
+
+
+def space_drawer(model, agent_portrayal):
+    # Ensure previously opened figures are closed to manage memory
+    plt.close("all")
+    # Smaller figure size
+    fig, ax = plt.subplots(figsize=(8, 5), dpi=100)  # Adjust figure size here
```

Comment:
You have to avoid using global Matplotlib functions at any time when using Solara. See https://github.com/projectmesa/mesa-examples/blob/1933cbc8251ca047273e0f88193f692ccbb86de6/examples/sugarscape_g1mt/app.py#L32-L33 for how to do it. 

See https://matplotlib.org/stable/users/explain/axes/tight_layout_guide.html for how to do tight layout without using the `plt.*` API.

Reply:
@rht Adopted to Figure as in the example by removing plt